docs/configuring-a-resource.md

```diff
@@ -0,0 +1,410 @@
+doing the same as above:
+
+```go
+func Configure(p *config.Provider) {
+    p.AddResourceConfigurator("aws_vpc", func (r *config.Resource) {
+        r.ExternalName = config.IdentifierFromProvider
+    })
+}
+```
+
+Let's check another resource, [aws_s3_bucket] which requires some other
+configuration. Reading the [import section of s3 bucket] we see that bucket is
+imported with its **name** which is provided with the [bucket] argument.
+We can just use the CR name as the bucket name, and we don't have to disable
+name initializer as we did above.
+
+However, since we are using metadata name as `bucket` argument, we need the
+following two:
+
+- Fill `bucket` attribute using external-name annotation, so that Terraform
+  knows the value we want to provide:
+
+    ```go
+    SetIdentifierArgumentFn: func(base map[string]interface{}, name string) {
+    	base["bucket"] = name
+    },
+    ```
+
+- Omit `bucket` and `bucket_prefix` from the crd spec, so that we don't have
+  multiple inputs for the same thing (name of the bucket):
+    ```go
+    OmittedFields: []string{
+        "bucket",
+        "bucket_prefix",
+    },
+    ```
+
+Hence, we end up the following external name configuration for `aws_s3_bucket`
+resource:
+
+```go
+func Configure(p *config.Provider) {
+    p.AddResourceConfigurator("aws_s3_bucket", func(r *config.Resource) {
+        r.ExternalName = config.ExternalName{
+            SetIdentifierArgumentFn: func(base map[string]interface{}, name string) {
+                base["bucket"] = name
+            },
+            OmittedFields: []string{
+                "bucket",
+                "bucket_prefix",
+            },
+        }
+    })
+}
+```
+
+Please note, you can always check resource configurations of existing Providers
+as further examples under `config/<group>/config.go`.
+
+### Cross Resource Referencing
+
+Crossplane uses cross resource referencing to [handle dependencies] between
+managed resources. For example, if you have an iam User defined as a Crossplane
+managed resource, and you want to create an Access Key for that user, you would
+need to refer to the User CR from the Access Key resource. This is handled by
+cross resource referencing.
+
+See how the [user] referenced at `forProvider.userRef.name` field of the
+Access Key in the following example:
+
+```yaml
+apiVersion: iam.aws.tf.crossplane.io/v1alpha1
+kind: User
+metadata:
+  name: sample-user
+spec:
+  forProvider: {}
+---
+apiVersion: iam.aws.tf.crossplane.io/v1alpha1
+kind: AccessKey
+metadata:
+  name: sample-access-key
+spec:
+  forProvider:
+    userRef:
+      name: sample-user
+  writeConnectionSecretToRef:
+    name: sample-access-key-secret
+    namespace: crossplane-system
+```
+
+Historically, reference resolution method were written by hand which requires
+some effort, however, with the latest Crossplane code generation tooling, it is
+now possible to [generate reference resolution methods] by just adding some
```

Comment:
nitpick: not sure how it used to be recently is relevant since this guide will live months/years.

Reply:
I would say it is still recent :) We can reword that paragraph in a later revisit.